### PR TITLE
fix: correct handling of keepdims and mask_identity for weighted mean

### LINKED ISSUE #3285: Row-wise weighted mean gives incorrect results
### Version of Awkward Array

2.6.9

### Description and code to reproduce

I am doing a row-wise weighted mean in an awkward array and getting wrong results.

Here's an MRE with outputs in the comments:

```python
import awkward as ak

data = ak.Array(
    [
        [1, 2, 3],
        [4, 5],
    ]
)

weight = ak.Array(
    [
        [1, 1, 2],
        [1, 10],
    ]
)

# manual row-by-row - expected results
print(ak.mean(data[0], weight=weight[0]))  # -> 2.25
print(ak.mean(data[1], weight=weight[1]))  # -> 4.909090909090909

# manual vectorized - expected results
weights_norm = weight / ak.sum(weight, axis=1)
print(ak.sum(weights_norm * data, axis=1))  # -> [2.25, 4.91]

# the most natural call I expected to work - incorrect result in the 2nd row
print(ak.mean(data, weight=weight, axis=1))  # -> [2.25, 13.5]
```

### PR DESCRIPTION
**Pull request by nj-vs-vh**
Fixes #3285

### CODE REVIEW

File: src/awkward/operations/ak_mean.py [hunk 1]
```diff
@@ -225,8 +225,8 @@ def _impl(x, weight, axis, keepdims, mask_identity, highlevel, behavior, attrs):
             sumw = ak.operations.ak_sum._impl(
                 x * 0 + weight,
                 axis,
-                keepdims,
-                mask_identity,
+                keepdims=True,
+                mask_identity=True,
```
Comment (nj-vs-vh) [inline src/awkward/operations/ak_mean.py:229]: This wasn't in my original issue, but was brought up by @ianna in the discussion, so I decided to fix it as well; also added tests for it

**Review by ianna — APPROVED** [review]
@nj-vs-vh - Looks great! Thanks for finding the bug and fixing it!

**Review by jpivarski — APPROVED** [review]
Thank you very much for this important fix!

**Comment by jpivarski** [comment]
@all-contributors please add @nj-vs-vh for code

**Pull request merged**